python/GangaDirac/Lib/Backends/DiracBase.py

```diff
@@ -483,9 +520,30 @@ def debug(self):
             logger.error(result.get('Message', ''))
 
     @staticmethod
-    def _getStateTime(job, status):
+    def _bulk_updateStateTime( jobStateDict ):
+        """ This performs the same as the _getStateTime method but loops over a list of job ids within the DIRAC namespace (much faster)
```

Comment:
It was up until I discovered a bug about 15min ago, as it didn't change the structure of the PR I was keen to hear feedback which will be implemented before looking to merge again